ide/db.sql.editor/nbproject/project.xml

```diff
@@ -1,341 +1,364 @@
-                    </run-dependency>
-                </dependency>
-            </module-dependencies>
-            <test-dependencies>
-                <test-type>
-                    <name>unit</name>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.libs.junit4</code-name-base>
-                        <compile-dependency/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.csl.api</code-name-base>
-                        <recursive/>
-                        <compile-dependency/>
-                        <test/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.db</code-name-base>
-                        <compile-dependency/>
-                        <test/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.db.metadata.model</code-name-base>
-                        <compile-dependency/>
-                        <test/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.editor.mimelookup</code-name-base>
-                        <compile-dependency/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.editor.mimelookup.impl</code-name-base>
-                        <compile-dependency/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.lexer</code-name-base>
-                        <compile-dependency/>
-                        <test/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.nbjunit</code-name-base>
-                        <recursive/>
-                        <compile-dependency/>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.netbeans.modules.projectapi.nb</code-name-base>
-                    </test-dependency>
-                    <test-dependency>
-                        <code-name-base>org.openide.util.lookup</code-name-base>
-                        <compile-dependency/>
-                        <test/>
-                    </test-dependency>
-                </test-type>
-            </test-dependencies>
-            <friend-packages>
-                <friend>com.sun.tools.odb</friend>
-                <friend>org.netbeans.modules.php.editor</friend>
-                <package>org.netbeans.modules.db.sql.editor.api.completion</package>
-            </friend-packages>
-        </data>
-    </configuration>
-</project>
+<?xml version="1.0" encoding="UTF-8"?>

```

Comment:
I checked in with Netbeans git client. I'll see if I missed a lf/cr config option there.  Although I may have checked out with external git client at an earlier point.  Should the git in Netbeans default to no change in line feed format?

Reply:
For the linefeed issues: Depends on your environment.
For the build errors: No, most probably the build is just flaky, you need to check the errors.

Reply:
As part of recent check in, I believe I've resolved this issue.  Please confirm when able

> I think this needs cleanup before it can be merged. Primarily only changes, that should be in master (and are working) should be in this PR. I left some comments inline, which might help.
> 
> My concern right now, that this opens a completly new can of worms, because the core assumption ("real" completion only happens when a connection is available) is no broken. I saw some guards added, but yet I managed to get this:
> 
> ```
> java.lang.NullPointerException
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.completeSimpleIdentBasedOnFromClause(SQLCompletionQuery.java:683)
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.completeColumnWithDefinedTuple(SQLCompletionQuery.java:501)
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.completeSelect(SQLCompletionQuery.java:268)
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.doQuery(SQLCompletionQuery.java:211)
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.doQuery(SQLCompletionQuery.java:152)
> 	at org.netbeans.modules.db.sql.editor.completion.SQLCompletionQuery.query(SQLCompletionQuery.java:101)
> 	at org.netbeans.spi.editor.completion.support.AsyncCompletionTask.run(AsyncCompletionTask.java:198)
> 	at org.openide.util.RequestProcessor$Task.run(RequestProcessor.java:1418)
> 	at org.netbeans.modules.openide.util.GlobalLookup.execute(GlobalLookup.java:45)
> 	at org.openide.util.lookup.Lookups.executeWith(Lookups.java:278)
> 	at org.openide.util.RequestProcessor$Processor.run(RequestProcessor.java:2033)
> Caused: org.openide.util.RequestProcessor$FastItem: task failed due to
> ```
> 
> from a simple query (completion was invoked at |):
> 
> ```
> select * from x WHERE |;
> ```

Reply:
Unless otherwise concerned...close out this review item.